### PR TITLE
Us119012/overdue refactor

### PR DESCRIPTION
**Pull request by johngwilkinson**
Regression testing: local e2e; with dummy data; apply filter alone and in combination with others

FYI @anhill-D2L @MykolaGalian @NicholasHallman @Vitalii-Misechko

### CODE REVIEW

File: components/overdue-assignments-card.js [hunk 1]
```diff
@@ -32,7 +46,17 @@ class OverdueAssignmentsCard extends SkeletonMixin(Localizer(MobxLitElement)) {
...
 
 	get _cardValue() {
-		return this.data.usersCountsWithOverdueAssignments;
+		return this.data.getRecordsInView(filterId)
+			.reduce((acc, record) => {
+				if (!acc.has(record[RECORD.USER_ID]) && record[RECORD.OVERDUE] !== 0) {
+					acc.add(record[RECORD.USER_ID]);
+				}
+				return acc;
+			}, 	new Set()).size;
```
Comment (bemailloux) [inline components/overdue-assignments-card.js:55]: `Out of scope / optional` Readability improvements
```
const usersWithOverdueAssignments = new Set(...this.data.getRecordsInView(filterid)
  .filter(record => record[RECORD.OVERDUE] > 0)
  .map(record => record[RECORD.USER_ID])
);
return usersWithOverdueAssignments.size;
```

File: test/model/data.test.js [hunk 2]
```diff
@@ -358,13 +358,6 @@ describe('Data', () => {
 		});
 	});
 
-	describe('overdueAssignments', () => {
-		it('should return a number of users with overdue assignments', async() => {
```
Comment (bemailloux) [inline test/model/data.test.js:362]: Shouldn't there be a corresponding addition for this deletion? I don't see one

Comment (johngwilkinson) [reply test/model/data.test.js:362]: It's sneaky, but it's there: the overdue-assignments-card tests now use the records this test was using - and happened to already expect the same count.

**Review by bemailloux — APPROVED** [review]

**Review by rohitvinnakota — APPROVED** [review]

File: components/overdue-assignments-card.js (hunk 1 above)
Comment (Vitalii-Misechko) [reply components/overdue-assignments-card.js:55]: Interesting, passing `filterId` into `getRecordsInView` we exclude `OverdueAssignmentsFilter`  effect on records. Then we manually apply `OverdueAssignmentsFilter` here. Should we just contract @bemailloux's code? to 
```JS
const usersWithOverdueAssignments = new Set(...this.data.getRecordsInView()
  .map(record => record[RECORD.USER_ID])
);
return usersWithOverdueAssignments.size;
```

**Pull request merged**